### PR TITLE
Fix dataframe loading bug

### PR DESCRIPTION
**Pull request by tavinathanson**
This does warrant a test. I'll write one up.

### CODE REVIEW

File: cohorts/load.py [hunk 1]
```diff
@@ -267,8 +267,14 @@ def as_dataframe(self, join_with=None, join_how=None):
         if type(join_with) == str:
             join_with = [join_with]
 
+        # Convert strings to DataFrameLoader objects
+        df_loaders = []
+        for df_loader in self.df_loaders:
+            if df_loader.name in join_with:
+                df_loaders.append(df_loader)
+
```
Comment (arahuja) [inline cohorts/load.py:272]: This could be 

```
df_loaders = [df_loader for df_loader in self.df_loaders if df_loader.name in join_with]
```

**Comment by arahuja** [comment]
I'm not sure I know the bug here? And why the default changed to `inner` from `outer`

**Comment by tavinathanson** [comment]
@arahuja the bug was that I was passing strings along to code that expected `DataFrameLoader` objects. Re `outer` to `inner`: I was being cautious with `outer` as the default, but it actually creates problems; namely, since outer joined patients don't have `OS` and `PFS`, they fail the `OS`/`PFS` verification. I do print the following when the join happens, though: `inner join with pdl1: 25 to 25 rows`

**Comment by arahuja** [comment]
👍

**Pull request merged**